crates/ef-testing/src/models/case.rs

```diff
@@ -116,23 +64,11 @@ impl BlockchainTestCase {
         Ok(())
     }
 
-    async fn handle_transaction(
-        &self,
-        sequencer: &mut KakarotSequencer,
-        test_case_name: &str,
-    ) -> Result<(), RunnerError> {
-        let test = self.test(test_case_name)?;
-
+    fn handle_transaction(&self, sequencer: &mut KakarotSequencer) -> Result<(), RunnerError> {
         // we extract the transaction from the block
-        let block = test
-            .blocks
-            .first()
-            .ok_or_else(|| RunnerError::Other("test has no blocks".to_string()))?;
+        let block = &self.block;
         // we adjust the rlp to correspond with our currently hardcoded CHAIN_ID
```

Comment:
I can update the KKT0 to have the chain id set in constructor and in a storage var

Reply:
and then the EOA would make a call to it to get the chain id?

Reply:
why not in the initialize, let's do this in another PR